community/codegen/src/main/java/org/neo4j/codegen/bytecode/ByteCodeExpressionVisitor.java

```diff
@@ -0,0 +1,542 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.codegen.bytecode;
+
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+
+import java.lang.reflect.Modifier;
+
+import org.neo4j.codegen.Expression;
+import org.neo4j.codegen.ExpressionVisitor;
+import org.neo4j.codegen.FieldReference;
+import org.neo4j.codegen.LocalVariable;
+import org.neo4j.codegen.MethodReference;
+import org.neo4j.codegen.TypeReference;
+
+import static org.neo4j.codegen.ByteCodeUtils.byteCodeName;
```

Comment:
Sure we could do that. I am not sure I see the point though, since input input is typically a `TypeReference` say a `String`-reference and output is the "bytecode type", for example  `Ljava/lang/String`. To me mocking this would to me be a little like mocking `java.lang.Math.abs` - you could do it but what is the point, when do you want to mock a function to return the incorrect output given an input?